src/storage-types/src/sources.rs

```diff
@@ -1521,6 +1529,54 @@ impl<C: ConnectionAccess> SourceConnection for KafkaSourceConnection<C> {
             })
             .collect()
     }
+
+    fn alter_compatible(&self, id: GlobalId, other: &Self) -> Result<(), StorageError> {
+        if self == other {
+            return Ok(());
+        }
+
+        let KafkaSourceConnection {
+            // Connection details may change
+            connection: _,
+            connection_id,
+            topic,
+            start_offsets,
+            group_id_prefix,
+            environment_id,
+            metadata_columns,
+            connection_options,
+        } = self;
+
+        let compatibility_checks = [
+            (connection_id == &other.connection_id, "connection_id"),
+            (topic == &other.topic, "topic"),
+            (start_offsets == &other.start_offsets, "start_offsets"),
+            (group_id_prefix == &other.group_id_prefix, "group_id_prefix"),
+            (environment_id == &other.environment_id, "environment_id"),
+            (
+                metadata_columns == &other.metadata_columns,
+                "metadata_columns",
+            ),
+            (
+                connection_options == &other.connection_options,
+                "connection_options",
+            ),
+        ];
+
+        for (compatible, field) in compatibility_checks {
+            if !compatible {
+                tracing::warn!(
+                    "KafkaSourceConnection incompatible at {field}:\nself:\n{:#?}\n\nother\n{:#?}",
+                    self,
+                    other
+                );
+
+                return Err(StorageError::InvalidAlterSource { id });
+            }
+        }
+
+        Ok(())
```

Comment:
Uncovered

Reply:
This is expected because we don't support actually modifying any of these fields, so the function will always early exit with an `Ok(())`.